parser/agent.go

```diff
@@ -107,6 +109,71 @@ func InstrumentMain(manager *InstrumentationManager, c *dstutil.Cursor) {
 	}
 }
 
+func CheckForExistingApplicationInFunctions(manager *InstrumentationManager, decl *dst.FuncDecl, c *dstutil.Cursor) {
```

Comment:
Currently this function is only called in `InstrumentMain`. Do we expect to use this function elsewhere, or can we make it lowercase to avoid exporting?